### PR TITLE
Different ColumnDefinition for different platforms

### PR DESCRIPTION
**Pull request by NSzemenyei**
Hello @rbygrave ,

I opened a PR for the issue https://github.com/ebean-orm/ebean/issues/2559

Can you please check it?

Best wishes
Noemi

### CODE REVIEW

File: ebean-test/src/test/java/misc/migration/v1_0/EBasic.java [hunk 1]
```diff
@@ -41,6 +42,9 @@
 
   @Size(max=127)
   String description;
+  
+  @Column(columnDefinition = "db2;blob(64M);sqlserver,h2;varchar(227);varchar(127)")
+  String description2;
```
Comment (rPraml) [inline ebean-test/src/test/java/misc/migration/v1_0/EBasic.java:47]: Please add the same in v1_2. (or only in v1_1) The idea is to check both directions. The models in v1_0 and v1_2 should be equal

File: ebean-ddl-generator/src/main/java/io/ebeaninternal/dbmigration/ddlgeneration/platform/PlatformDdl.java [hunk 2]
```diff
@@ -317,12 +319,32 @@ public String convert(String type) {
...
+    assert tmp.length % 2 == 1;
+    for (int i = 0; i < tmp.length - 2; i+=2) {
+      String[] platforms = tmp[i].split(",");
+      for (String plat : platforms) {
+        if (platform.isPlatform(Platform.valueOf(plat.toUpperCase(Locale.ENGLISH)))) {
+          return tmp[i+1];
+        }
+      }
+    }
+    return tmp[tmp.length-1]; // else
```
Comment (rPraml) [inline ebean-ddl-generator/src/main/java/io/ebeaninternal/dbmigration/ddlgeneration/platform/PlatformDdl.java:345]: reformat/indent whole block

File: ebean-ddl-generator/src/main/java/io/ebeaninternal/dbmigration/ddlgeneration/platform/PlatformDdl.java [hunk 3]
```diff
@@ -317,12 +319,32 @@ public String convert(String type) {
...
     }
     return typeConverter.convert(type);
   }
 
+  // if columnType is different for different platforms, use pattern
+  // @Column(columnDefinition = PLATFORM1;DEFINITION1;PLATFORM2;DEFINITON2;DEFINITON-DEFAULT)
+  // e.g. @Column(columnDefinition = "db2;blob(64M);sqlserver,h2;varchar(227);varchar(127)")
+  private String extract(String type) {
+    String[] tmp = type.split(";");
+    assert tmp.length % 2 == 1;
```
Comment (rPraml) [inline ebean-ddl-generator/src/main/java/io/ebeaninternal/dbmigration/ddlgeneration/platform/PlatformDdl.java:336]:
```suggestion
    assert tmp.length % 2 == 1 : "You need an odd number of arguments. See Issue #2559 for details";
```
maybe throw an IAE (asserts may be disabled)

**Comment by NSzemenyei** [comment]
@rbygrave  we implemented also, if the columnDefinition ends with a semicolon, the not given platform types will be calculated from ebean:

`@Column(columnDefinition = "db2;blob(64M);")` means: blob(64M) for db2, other platforms will be determined in `BeanProperty.renderDbType`

**Pull request merged**